ros/src/computing/planning/common/lib/openplanner/op_planner/include/op_planner/BehaviorPrediction.h

```diff
@@ -0,0 +1,882 @@
+	double acl_w_max;
+	double ind_w_max;
+
+	double pose_w_min;
+	double dir_w_min;
+	double vel_w_min;
+	double acl_w_min;
+	double ind_w_min;
+
+	int n_stop;
+	int n_yield;
+	int n_left_branch;
+	int n_right_branch;
+
+	double p_stop;
+	double p_yield;
+	double p_left_branch;
+	double p_right_branch;
+
+	virtual ~ObjParticles()
+	{
+		DeleteTheRest(m_TrajectoryTracker);
+		m_TrajectoryTracker_temp.clear();
+	}
+
+	ObjParticles()
+	{
+		m_PredictionTime = 0;
+		best_beh_track = nullptr;
+		i_best_track = -1;
+		all_w = 0;
+		pose_w_t = 0;
+		dir_w_t = 0;
+		vel_w_t = 0;
+		acl_w_t = 0;
+		ind_w_t = 0;
+		max_w = DBL_MIN;
+		min_w = DBL_MAX;
+		max_w_raw = DBL_MIN;
+		min_w_raw = DBL_MAX;
+		avg_w = 0;
+
+		pose_w_max = -999999;
+		dir_w_max=-999999;
+		vel_w_max=-999999;
+		acl_w_max=-999999;
+		ind_w_max=-999999;
+
+		pose_w_min=999999;
+		dir_w_min=999999;
+		vel_w_min=999999;
+		acl_w_min=999999;
+		ind_w_min=999999;
+
+		n_stop = 0;
+		n_yield = 0;
+		n_left_branch = 0;
+		n_right_branch = 0;
+
+		p_stop = 0;
+		p_yield = 0;
+		p_left_branch = 0;
+		p_right_branch = 0;
+	}
+
+//	void CalculateProbabilities()
+//	{
+//		for(unsigned int i = 0; i < m_TrajectoryTracker.size(); i++)
+//		{
+//			m_TrajectoryTracker.at(i)->CalcProbabilities();
+//		}
+//
+//		if(m_TrajectoryTracker.size() > 0)
+//		{
+//			best_beh_track = m_TrajectoryTracker.at(0);
+//			i_best_track = 0;
+//		}
+//
+//		for(unsigned int i = 1; i < m_TrajectoryTracker.size(); i++)
+//		{
+//			if(m_TrajectoryTracker.at(i)->best_p > best_beh_track->best_p)
+//			{
+//				best_beh_track = m_TrajectoryTracker.at(i);
+//				i_best_track = i;
+//			}
+//		}
+//	}
+
+	class LLP
+	{
+	public:
+		int new_index;
+		double match_percent;
+		TrajectoryTracker* pTrack;
+
+		LLP()
+		{
+			new_index = -1;
+			match_percent = -1;
+			pTrack = 0;
+
+		}
+	};
+
+	void DeleteFromList(std::vector<TrajectoryTracker*>& delete_me_track, TrajectoryTracker* track)
+	{
+		for(unsigned int k = 0; k < delete_me_track.size(); k++)
+		{
+			if(delete_me_track.at(k) == track)
+			{
+				delete_me_track.erase(delete_me_track.begin()+k);
+				return;
+			}
+		}
+	}
+
+	void DeleteTheRest(std::vector<TrajectoryTracker*>& delete_me_track)
+	{
+		for(unsigned int k = 0; k < delete_me_track.size(); k++)
+		{
+			delete delete_me_track.at(k);
+		}
+
+		delete_me_track.clear();
+	}
+
+	static bool IsBeggerPercentage(const LLP& p1, const LLP& p2)
+	{
+		return p1.match_percent > p2.match_percent;
+	}
+
+	void MatchWithMax(std::vector<LLP>& matching_list, std::vector<TrajectoryTracker*>& delete_me_track, std::vector<TrajectoryTracker*>& check_list)
+	{
+		if(matching_list.size() == 0 ) return;
+
+		std::sort(matching_list.begin(), matching_list.end(), IsBeggerPercentage);
+
+		while(matching_list.size()>0)
+		{
+			LLP f = matching_list.at(0);
+			f.pTrack->UpdatePathAndIndex(obj.predTrajectories.at(f.new_index), f.new_index);
+
+			bool bFound = false;
+			for(unsigned int k=0; k < check_list.size(); k++)
```

Comment:
personal preference

Reply:
@hatem-darweesh NP. But std::find can reduce the number of lines 1/9 (one ninth) where we develop, review, and maintain. If there are ten loops finding an element, it can reduce 81 (= 90 - 9) lines in the Autoware code base. If you like it, please use it :)

- The number of lines of this loop is 9 (from Line 688 to 695).
- The number of lines of std::find is 1.